plugins/aws/ec2/outdatedAmi.js

```diff
@@ -0,0 +1,125 @@
+                ['ec2', 'describeImages', region]);
+
+            if (!describeImages) return rcb();
+
+            if (describeImages.err || !describeImages.data) {
+                helpers.addResult(results, 3,
+                    `Unable to query for AMIs: ${helpers.addError(describeImages)}`,
+                    region);
+                return rcb();
+            }
+
+            if (!describeImages.data.length) {
+                helpers.addResult(results, 0,
+                    'No Amazon Machine Images found', region);
+                return rcb();
+            }
+
+            var describeInstances = helpers.addSource(cache, source,
+                ['ec2', 'describeInstances', region]);
+
+            var describeLaunchConfigurations = helpers.addSource(cache, source,
+                ['autoscaling', 'describeLaunchConfigurations', region]);
+
+            if (!describeInstances || describeInstances.err || !describeInstances.data) {
+                helpers.addResult(results, 3,
+                    `Unable to query EC2 instances: ${helpers.addError(describeInstances)}`, region);
+                return rcb();
+            }
+
+            
+            if (!describeLaunchConfigurations || describeLaunchConfigurations.err || !describeLaunchConfigurations.data) {
+                helpers.addResult(results, 3,
+                    `Unable to query Auto Scaling launch configurations: ${helpers.addError(describeLaunchConfigurations)}`, region);
+                return rcb();
+            }
+            let found = false;
+            describeImages.data.forEach(image => {
+                const currentDate = new Date();
+                if (image.DeprecationTime && new Date(image.DeprecationTime) < currentDate) {                    
+                    found = true;
+                    if (config.check_ami_usage){
+                        let usedAmi = false;
+                        if (describeInstances.data.length) {
+                            describeInstances.data.forEach(instance => {
+                                if (instance.Instances && instance.Instances.length) {
+                                    instance.Instances.forEach(element => {
+                                        if (element.ImageId && image.ImageId == element.ImageId) {
+                                            usedAmi = true;
+                                        }
+                                    });
+                                }
+                            });
+                        }
+            
+                        if (describeLaunchConfigurations.data.length) {
+                            describeLaunchConfigurations.data.forEach(config => {
+                                if (config.ImageId && image.ImageId == config.ImageId) {
+                                    usedAmi = true;
+                                }
+                            });
+                        }
+    
```

Comment:
3 cases.
1. inuse + deprecated => fail
2. deprecated + notinuse => pass
3. not deprecated => pass